core/module.h

```diff
@@ -385,22 +367,35 @@ class Module {
 
   virtual CheckConstraintResult CheckModuleConstraints() const;
 
-  /*!
-   * Whether the module overrides RunTask.
-   */
-  virtual bool IsTask() const { return false; };
+  // For testing.
+  int children_overload() const { return children_overload_; };
+  const std::vector<Module *> &parent_tasks() const { return parent_tasks_; };
 
-  int children_overload() { return children_overload_; };  // For testing.
+  // Signals to parent task(s) that module is overloaded.
+  void SignalOverload() {
+    if (overload_) {
```

Comment:
You may want to add a TODO somewhere that overload and underload signaling only are safe if the module that they are called from is *not* marked as thread safe, indicating that it cannot be invoked by more than one worker.  If more than one worker can call a module then these methods would have to be changed to be thread safe.